hub/src/server/server.js

```diff
@@ -7,22 +7,28 @@ import { ProofChecker } from './ProofChecker'
 import type { Readable } from 'stream'
 import type { DriverModel } from './driverModel'
 
+export type HubServerConfig = {
+  whitelist?: Array<string>, 
+  serverName?: string, 
+  readURL?: string, 
+  requireCorrectHubUrl?: boolean, 
+  validHubUrls?: Array<string> 
+}
+
 export class HubServer {
   driver: DriverModel
   proofChecker: ProofChecker
-  whitelist: Array<string>
+  whitelist: ?Array<string>
   serverName: string
   readURL: ?string
   requireCorrectHubUrl: boolean
   validHubUrls: ?Array<string>
-  constructor(driver: DriverModel, proofChecker: ProofChecker,
-              config: { whitelist: Array<string>, serverName: string,
-                        readURL?: string, requireCorrectHubUrl?: boolean,
-                        validHubUrls?: Array<string> }) {
+
+  constructor(driver: DriverModel, proofChecker: ProofChecker, config: HubServerConfig) {
     this.driver = driver
     this.proofChecker = proofChecker
     this.whitelist = config.whitelist
-    this.serverName = config.serverName
+    this.serverName = (config.serverName: any)
```

Comment:
why does serverName need to be `any` ?

Reply:
Existing test code is dependent on `serverName` coercing to `"undefined"`.
Perhaps we can update that test code to not depend on that instead?

Reply:
Ah, yeah, let's do that